packages/ui/components/navigation/tabs/navigation.test.tsx

```diff
@@ -0,0 +1,147 @@
+        } else {
+          expect(spanSibling).toBeFalsy();
+        }
+      });
+    });
+
+    test("Should render actions correctly", () => {
+      const handleClick = vi.fn();
+      const mockActions = <button onClick={handleClick}>Actions</button>;
+      render(<HorizontalTabs tabs={mockTabs} actions={mockActions} />);
+      const actionsElement = screen.getByText("Actions");
+      expect(actionsElement).toBeInTheDocument();
+      fireEvent.click(actionsElement);
+
+      expect(handleClick).toHaveBeenCalled();
+    });
+  });
+
+  describe("Test VerticalTabs Component", () => {
+    const mockTabs = [
+      {
+        name: "Tab 1",
+        href: "/tab1",
+        disableChevron: true,
+        disabled: true,
+        icon: PlusIcon,
+      },
+      { name: "Tab 2", href: "/tab2", isExternalLink: true },
+      { name: "Tab 3", href: "/tab3", info: "info" },
+    ];
+
+    beforeEach(() => {
+      vi.clearAllMocks();
+    });
+
+    test("Should render tabs with correct name and href", () => {
+      render(<VerticalTabs tabs={mockTabs} />);
+      mockTabs.forEach((tab) => {
+        const tabLabelElement = screen.getByTestId(`vertical-tab-${tab.name}`);
+        expect(tabLabelElement).toBeInTheDocument();
+
+        const name = screen.getByText(tab.name);
+        expect(name).toBeInTheDocument();
+        expect(tabLabelElement).toHaveAttribute("href", tab.href);
+      });
+    });
+
+    test("Should render correctly if props are passed", () => {
+      render(<VerticalTabs tabs={mockTabs} />);
+
+      const iconElement = screen.getAllByTestId("icon-component");
+      const externalLink = screen.getAllByTestId("external-link");
+      const chevronRight = screen.getAllByTestId("chevron-right");
+
+      mockTabs.forEach((tab) => {
+        const tabName = screen.getByText(tab.name);
+        expect(tabName).toBeInTheDocument();
+
+        const aTag = screen.getByTestId(`vertical-tab-${tab.name}`);
+        const tabContainer = tabName.closest("a");
+        const infoElement = tabContainer?.querySelector("p[title='info']");
+
+        expect(chevronRight.length).toEqual(mockTabs.length - 1);
+        if (tab.disabled) {
+          expect(aTag).toHaveClass("pointer-events-none !opacity-30");
+        } else {
+          expect(aTag).not.toHaveClass("pointer-events-none !opacity-30");
+        }
+
+        if (tab.info) {
+          expect(infoElement).toBeInTheDocument();
+        } else {
+          expect(infoElement).toBeNull();
+        }
+
+        if (tab.isExternalLink) {
+          expect(aTag).toHaveAttribute("target", "_blank");
+        } else {
+          expect(aTag).toHaveAttribute("target", "_self");
+        }
+      });
+
+      expect(externalLink.length).toEqual(1);
+      expect(iconElement.length).toEqual(1);
+
+      const svg = document.querySelectorAll("svg");
+      const svgLength = svg?.length;
+      expect(svgLength).toEqual(mockTabs.length + 1);
```

Comment:
Why would there be 1 svg more than tabs ? This isn't clear here.
From a user perspective I would see an icon along with every item. So, we should check the presence per item and not in total.

Also one additional SVG is probably for chevron which should be checked for it's presence in appropriate place.